src/db.rs

```diff
@@ -38,315 +40,426 @@ impl Default for Connection {
-    let keyset_json_str =
-        keyset_json_opt.ok_or_else(|| anyhow!("db::get_keyset_ids- keyset id not found"))?;
+pub async fn get_keyset_ids(keysetid: &str) -> Result<Vec<String>> {
+    let dbpool = get_dbpool().await?;
 
-    let rks: Vec<String> = serde_json::from_str(&keyset_json_str).unwrap();
+    let query_str = "SELECT kskeys FROM keysets WHERE keysetid = ?";
+    let new_query_str = replace_binds(dbpool.any_kind(), query_str);
 
+    let keyset_row = sqlx::query(&new_query_str)
+        .bind(keysetid)
+        .fetch_one(&dbpool)
+        .await?;
+    let rks: Vec<String> = serde_json::from_str(&keyset_row.try_get::<String, _>(0)?).unwrap();
     Ok(rks)
 }
 
-pub fn get_secret(secret_id: &str) -> Result<request::Key> {
-    let mut dbconn = get_dbconn()?;
-    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
+pub async fn get_secret(secret_id: &str) -> Result<request::Key> {
+    let dbpool = get_dbpool().await?;
 
-    let mqstr = "SELECT secret FROM secrets WHERE secret_id = ?";
-    let val: Option<String> = trnsx.exec_first(mqstr, (secret_id,))?;
-
-    let payload = val.ok_or_else(|| anyhow!("secret not found."))?;
+    let query_str = "SELECT secret FROM secrets WHERE secret_id = ?";
+    let new_query_str = replace_binds(dbpool.any_kind(), query_str);
 
+    let secret_row = sqlx::query(&new_query_str)
+        .bind(secret_id)
+        .fetch_one(&dbpool)
+        .await?;
+    let secret = secret_row.try_get::<String, _>(0)?;
     Ok(request::Key {
         id: secret_id.to_string(),
-        payload,
+        payload: secret,
     })
 }
 
-pub fn insert_secret(secret_id: &str, secret: &str, policy_id: Option<u64>) -> Result<()> {
-    let mut dbconn = get_dbconn()?;
-
-    let mqstr = "INSERT INTO secrets (secret_id, secret, polid ) VALUES(?, ?, ?)";
-
-    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
-
-    trnsx.exec_drop(mqstr, (&secret_id, &secret, &policy_id))?;
-    trnsx.commit()?;
-    Ok(())
+pub async fn insert_secret(secret_id: &str, secret: &str, policy_id: Option<u64>) -> Result<()> {
+    let dbpool = get_dbpool().await?;
+
+    match policy_id {
+        Some(p) => {
+            let query_str = "INSERT INTO secrets (secret_id, secret, polid ) VALUES(?, ?, ?)";
+            let new_query_str = replace_binds(dbpool.any_kind(), query_str);
+            sqlx::query(&new_query_str)
+                .bind(secret_id)
+                .bind(secret)
+                .bind(p as i64)
+                .execute(&dbpool)
+                .await?;
+            Ok(())
+        }
+        None => {
+            let query_str = "INSERT INTO secrets (secret_id, secret) VALUES(?, ?)";
+            let new_query_str = replace_binds(dbpool.any_kind(), query_str);
+            sqlx::query(&new_query_str)
+                .bind(secret_id)
+                .bind(secret)
+                .execute(&dbpool)
+                .await?;
+            Ok(())
+        }
+    }
```

Comment:
I wonder if the following simpler version will work both on mysql and postgres (converting the `Option<u64>` to `Option<i64>`):

```rust
pub async fn insert_secret(secret_id: &str, secret: &str, policy_id: Option<u64>) -> Result<()> {
    let dbpool = get_dbpool().await?;
    let query_str = "INSERT INTO secrets (secret_id, secret, polid ) VALUES(?, ?, ?)";
    let new_query_str = replace_binds(dbpool.any_kind(), query_str);
    sqlx::query(&new_query_str)
        .bind(secret_id)
        .bind(secret)
        .bind(policy_id.map(|p| p as i64))
        .execute(&dbpool)
        .await?;
    Ok(())
```

Reply:
This does work with run_local_tests.sh on my laptop, so I have made this change.  This is a nice way to simplify this function.  I was concerned that postgres or sql would choke on a None from the policy_id option, but it does not appear to do so.  I think that we are good.  Thanks for suggesting this.